dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -554,14 +580,37 @@ export default defineComponent({
       try {
         this.postEuTaxonomyDataForFinancialsProcessed = false;
         this.messageCount++;
+        
+        // JSON.parse/stringify used to clone the formInputsModel in order to stop Proxy refreneces
+        const clonedFormInputsModel = JSON.parse(JSON.stringify(this.formInputsModel)) as ObjectType;
+        const kpiSectionsModel = (clonedFormInputsModel.data as ObjectType).kpiSectionsModel;
+
+        const eligibilityKpis = Object.keys(kpiSectionsModel)
```

Comment:
I find these both filterings happening here too complicated. Why not use one call to filter in which we check that the key is contained in a predefined array of expected values.